models/classes/metadata/MetadataLomService.php

```diff
@@ -39,13 +38,30 @@ public function addPropertiesToMetadataBlock(array $properties, DOMDocument $man
             $manifest->documentElement->appendChild($metadataBlock);
         }
 
-        $metadataBlock->item(0)
-            ->appendChild($manifest->createElement('imsmd:lom'))
-            ->appendChild($manifest->createElement('imsmd:metaMetadata'))
-            ->appendChild($manifest->createElement('extension'))
-            ->appendChild($manifest->createElement('customProperties'));
+        $customProperties = $this->getCustomProperties($manifest);
+
+        // If there is no custom properties means that it is initial loop and we need to create customProperties node
+        if (empty($customProperties)) {
+            $metadataBlock->item(0)
+                ->appendChild($manifest->createElement('imsmd:lom'))
+                ->appendChild($manifest->createElement('imsmd:metaMetadata'))
+                ->appendChild($manifest->createElement('extension'))
+                ->appendChild($manifest->createElement('customProperties'));
+        }
 
         foreach ($properties as $property) {
+            $propertyExists = false;
+            if (!empty($customProperties)) {
```

Comment:
It seems that it should not be empty already cause you created them above, no?

Reply:
`$this->getCustomProperties($manifest);` should return empty array on a first loop.